### PR TITLE
Specify the target store name to achieve a soup per store

### PR DESCRIPTION
**Pull request by trooper2013**
Changes related to the ability to specify StoreName in JS smartstore/smartsync apis. Added new Store related API(s).

### CODE REVIEW

File: gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js [hunk 1]
```diff
@@ -30,6 +30,8 @@ var SERVICE = "com.salesforce.smartstore";
 
 var exec = require("com.salesforce.util.exec").exec;
 
+var defaultStoreName = "defaultStore";
```
Comment (bhariharan) [inline gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js]: This has to match what's defined in `native`, correct? Can we remove this, so `native` is the single source of truth and we don't have to maintain them in both?

File: gen/plugins/com.salesforce/com.salesforce.plugin.smartsync.js [hunk 2]
```diff
@@ -29,79 +29,89 @@ var SALESFORCE_MOBILE_SDK_VERSION = "5.0.0";
 var SERVICE = "com.salesforce.smartsync";
 
 var exec = require("com.salesforce.util.exec").exec;
+var defaultStoreName = "defaultStore";
```
Comment (bhariharan) [inline gen/plugins/com.salesforce/com.salesforce.plugin.smartsync.js]: This has to match what's defined in `native`, correct? Can we remove this, so `native` is the single source of truth and we don't have to maintain them in both?

File: gen/plugins_with_define/com.salesforce/com.salesforce.plugin.smartstore.js [hunk 3]
```diff
@@ -31,6 +31,8 @@ var SERVICE = "com.salesforce.smartstore";
 
 var exec = require("com.salesforce.util.exec").exec;
 
+var defaultStoreName = "defaultStore";
```
Comment (bhariharan) [inline gen/plugins_with_define/com.salesforce/com.salesforce.plugin.smartstore.js]: This has to match what's defined in `native`, correct? Can we remove this, so `native` is the single source of truth and we don't have to maintain them in both?

File: gen/plugins_with_define/com.salesforce/com.salesforce.plugin.smartsync.js [hunk 4]
```diff
@@ -30,79 +30,89 @@ var SALESFORCE_MOBILE_SDK_VERSION = "5.0.0";
 var SERVICE = "com.salesforce.smartsync";
 
 var exec = require("com.salesforce.util.exec").exec;
+var defaultStoreName = "defaultStore";
```
Comment (bhariharan) [inline gen/plugins_with_define/com.salesforce/com.salesforce.plugin.smartsync.js]: This has to match what's defined in `native`, correct? Can we remove this, so `native` is the single source of truth and we don't have to maintain them in both?

File: libs/cordova.force.js [hunk 5]
```diff
@@ -532,6 +532,8 @@ cordova.define("com.salesforce.plugin.smartstore", function (require, exports, m
 
     var exec = require("com.salesforce.util.exec").exec;
 
+    var defaultStoreName = "defaultStore";
```
Comment (bhariharan) [inline libs/cordova.force.js]: This has to match what's defined in `native`, correct? Can we remove this, so `native` is the single source of truth and we don't have to maintain them in both?

File: libs/cordova.force.js [hunk 6]
```diff
@@ -1010,79 +1069,89 @@ cordova.define("com.salesforce.plugin.smartsync", function (require, exports, mo
     var SERVICE = "com.salesforce.smartsync";
 
     var exec = require("com.salesforce.util.exec").exec;
+    var defaultStoreName = "defaultStore";
```
Comment (bhariharan) [inline libs/cordova.force.js]: This has to match what's defined in `native`, correct? Can we remove this, so `native` is the single source of truth and we don't have to maintain them in both?

File: test/MockSmartStore.js [hunk 7]
```diff
@@ -818,19 +818,99 @@ var MockSmartStore = (function(window) {
...
+var globalStores = {};
+var userStores = {};
+
+var getStore = function (args) {
+   var isGlobal = args[0].isGlobalStore;
+   var storeName = args[0].storeName;
+   var store;
+
+   if(storeName == null)
+      storeName = "defaultStore";
```
Comment (bhariharan) [inline test/MockSmartStore.js]: This has to match what's defined in `native`, correct? Can we remove this, so `native` is the single source of truth and we don't have to maintain them in both?

File: gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js [hunk 8]
```diff
@@ -39,6 +41,15 @@ var SoupSpec = function (soupName, features) {
 };
 
 /**
+ * StoreConfig constructor
+ */
+var StoreConfig = function (storeName, isGlobal) {
+    this.storeName = storeName;
+    this.isGlobal = isGlobal;
```
Comment (trooper2013) [inline gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js]: Done.

File: test/SFSmartStoreTestSuite.js [hunk 9]
```diff
@@ -2001,11 +2001,184 @@ if (typeof SmartStoreTestSuite === 'undefined') {
...
+                QUnit.ok(true,"closeCursor ok");
                 self.finalizeTest();
             });
     };
 
-}
+    /**
+     * TEST testCreateMultipleUserStores
+     */
+    SmartStoreTestSuite.prototype.testCreateMultipleUserStores = function()  {
```
Comment (trooper2013) [inline test/SFSmartStoreTestSuite.js]: done.

File: gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js (hunk 8 above)
Comment (wmathurin) [inline gen/plugins/com.salesforce/com.salesforce.plugin.smartstore.js]: It's still isGlobal.

File: libs/cordova.force.js [hunk 10]
```diff
@@ -541,6 +542,15 @@ cordova.define("com.salesforce.plugin.smartstore", function (require, exports, m
     };
 
     /**
+     * StoreConfig constructor
+     */
+    var StoreConfig = function (storeName, isGlobal) {
+        this.storeName = storeName;
+        this.isGlobal = isGlobal;
```
Comment (wmathurin) [inline libs/cordova.force.js]: Should be isGlobalStore

File: test/MockSmartSyncPlugin.js [hunk 11]
```diff
@@ -250,18 +250,19 @@ var MockSmartSyncPlugin = (function(window) {
...
 
     // Return module
     return module;
 })(window);
 
-var mockSyncManager = new MockSmartSyncPlugin(false);
-var mockGlobalSyncManager = new MockSmartSyncPlugin(true);
+
+var mockSyncManager = new MockSmartSyncPlugin({'isGlobalStore': false});
+var mockGlobalSyncManager = new MockSmartSyncPlugin({'isGlobalStore': true});
```
Comment (wmathurin) [inline test/MockSmartSyncPlugin.js]: Shouldn't you have a SyncManagerMap similar to the StoreMap you had in MockSmartStore?

**Pull request merged**